### ISSUE TITLE
Caching as long as the installation step succeeds

### ISSUE THREAD

**Issue by muzimuzhi** [teatimeguest/setup-texlive-action#317]
Sometimes the job failed but the step that uses this action succeeded. Then currently no caching is stored, and when there is no existing cache, every following job runs will install the TeX Live from scratch before a successful run happens.

https://github.com/teatimeguest/setup-texlive-action/blob/f8403fbb8e38c28da59f550951b959f3bdf00608/action.yml#L72-L73

Maybe the post-step caching can happen as long as the installation step succeeds. It can immediately speed up possibly failed following runs and make users receive faster feedbacks from CI when debugging.

**Comment by teatimeguest** [comment]
The current behavior is designed to match the behavior of official actions such as `actions/setup-node`. However, since TeX Live takes longer for setup than other CLI environments, your suggestion seems quite reasonable.

So I am thinking of changing the default behavior to that and adding an environment variable (`SETUP_TEXLIVE_ACTION_NO_CACHE_ON_FAILURE` or something) for reverting to existing behavior. A little concerned if this should be considered a breaking change, but I am expecting it should be fine.

Please let me know if you have any suggestions.

**Comment by teatimeguest** [comment]
I think a workaround at this time is to create a separate job for setup (untested):
```yaml
jobs:
  ensure-cached:
    runs-on: ubuntu-latest
    steps:
      - uses: teatimeguest/setup-texlive-action@v3
  ci:
    needs: ensure-cached
    runs-on: ubuntu-latest
    steps:
      - uses: teatimeguest/setup-texlive-action@v3
      - ...
```

**Comment by teatimeguest** [comment]
This has been released in [v3.3.1](https://github.com/teatimeguest/setup-texlive-action/releases/tag/v3.3.1). Thanks for the suggestion!
